apis/fluentbit/v1alpha2/plugins/filter/log_to_metrics_types.go

```diff
@@ -0,0 +1,112 @@
+package filter
+
+import (
+	"fmt"
+	"github.com/fluent/fluent-operator/v2/apis/fluentbit/v1alpha2/plugins"
+	"github.com/fluent/fluent-operator/v2/apis/fluentbit/v1alpha2/plugins/params"
+)
+
+// +kubebuilder:object:generate:=true
+
+// The Log To Metrics Filter plugin allows you to generate log-derived metrics. <br />
+// **For full documentation, refer to https://docs.fluentbit.io/manual/pipeline/filters/log_to_metrics**
+type LogToMetrics struct {
+	plugins.CommonParams `json:",inline"`
+	// Defines the tag for the generated metrics record
+	Tag string `json:"tag,omitempty"`
+	// Optional filter for records in which the content of KEY matches the regular expression.
+	// Value Format: FIELD REGEX
+	Regex []string `json:"regex,omitempty"`
+	// Optional filter for records in which the content of KEY does not matches the regular expression.
+	// Value Format: FIELD REGEX
+	Exclude []string `json:"exclude,omitempty"`
+	// Defines the mode for the metric. Valid values are [counter, gauge or histogram]
+	Metric_mode string `json:"metric_mode,omitempty"`
+	// Sets the name of the metric.
+	Metric_name string `json:"metric_name,omitempty"`
+	// Namespace of the metric
+	Metric_namespace string `json:"metric_namespace,omitempty"`
+	// Sets a sub-system for the metric.
+	Metric_subsystem string `json:"metric_subsystem,omitempty"`
+	// Sets a help text for the metric.
+	Metric_description string `json:"metric_description,omitempty"`
+	// Defines a bucket for histogram
+	Bucket []float `json:"bucket,omitempty"`
```

Comment:
pls check the github action errors like below:
![image](https://github.com/user-attachments/assets/f886d8c1-1ad2-4cc6-ae79-5ac06c26aad1)

Reply:
https://github.com/fluent/fluent-operator/actions/runs/10398584230/job/28798064743?pr=1305

Reply:
Sure, I tried testing locally but there were different errors. I will check and update the PR

Reply:
Should use `[]string` type here. This is because the config map type is same as `regex` parameter, isn't it?

Reply:
Oh ok